### ISSUE TITLE
Nullness checker gives a false negative for NULL values in some implementations of Queues

### ISSUE THREAD

**Issue by rajatjain1997** [typetools/checker-framework#1144]
The current implementation of the checker verifies the code below to be error free, i.e. NullPointerExceptions should not exist at runtime.

        public class NullnessCheck {	
	        public static void main(String[] args) {
		        ConcurrentLinkedQueue<@Nullable Integer> clq = new ConcurrentLinkedQueue<>();
		        clq.add(null);
            }
        }

However, when compiled, the code produces following errors -
'
        Exception in thread "main" java.lang.NullPointerException
        at java.util.concurrent.ConcurrentLinkedQueue.checkNotNull(ConcurrentLinkedQueue.java:914)
	        at java.util.concurrent.ConcurrentLinkedQueue.offer(ConcurrentLinkedQueue.java:327)
	        at java.util.concurrent.ConcurrentLinkedQueue.add(ConcurrentLinkedQueue.java:297)
	        at Test.NullnessCheck.main(NullnessCheck.java:12)
'

The same also happens with ArrayDeque and all other queues that don't allow null objects.

This happens because the current implementation of the queue interface allows Nullable objects-
`public interface Queue<E extends @Nullable Object> extends Collection<E>`

However every implementation of the interface, like the one above does not allow @Nullable objects. This is an issue throughout the collections framework including maps and lists where the implementing classes can restrict @Nullable objects while the superclasses won't. These classes need to be implemented separately, to make the checker complete. The checker should give a warning when @Nullable annotations are applied to these types.

**Comment by wmdietl** [comment]
Isn't this just a duplicate of https://github.com/typetools/checker-framework/issues/1142 and the solution is to add non-null bounds for these concurrent collections?

**Comment by rajatjain1997** [comment]
@wmdietl I'm sorry, I read the topics of the issues and since I'm not familiar with ConcurrentHashMap, I though that the author is trying to state a missing annotated file (And I particularly did not pay attention because my bug was about collections in general and stemmed from queues), and thus did not realize how similar they were. Only after posting did I realize it. Can you please merge these? I'm sorry if I caused any inconvenience.

**Comment by mernst** [comment]
Duplicate of #1142.  This does give another important case that #1142 should handle.  Thanks for the report.
